supporter-product-data/src/main/scala/com/gu/lambdas/ProcessSupporterRatePlanItemLambda.scala

```diff
@@ -0,0 +1,132 @@
+      )
+    }
+    Await.result(futureProcessor, 10.seconds)
+  }
+}
+
+class ContributionProcessor(stage: Stage, config: ZuoraQuerierConfig) {
+  val contributionIds = ContributionIds.forStage(stage)
+  lazy val zuoraService = new ZuoraSubscriptionService(config, configurableFutureRunner(60.seconds))
+
+  def isRecurringContribution(supporterRatePlanItem: SupporterRatePlanItem) =
+    contributionIds.contains(supporterRatePlanItem.productRatePlanId)
+
+  def addAmountIfContribution(supporterRatePlanItem: SupporterRatePlanItem) =
+    if (isRecurringContribution(supporterRatePlanItem)) {
+      SafeLogger.info(s"Supporter ${supporterRatePlanItem.identityId} is a recurring contributor")
+      fetchContributionAmountFromZuora(supporterRatePlanItem)
+    } else
+      Future.successful(supporterRatePlanItem)
+
+  def fetchContributionAmountFromZuora(supporterRatePlanItem: SupporterRatePlanItem) =
+    zuoraService
+      .getSubscription(supporterRatePlanItem.subscriptionName)
+      .map { sub =>
+        SafeLogger.info(
+          s"Contribution amount for supporter ${supporterRatePlanItem.identityId} is ${sub.contributionAmount}",
+        )
+        supporterRatePlanItem.copy(contributionAmount = sub.contributionAmount)
+      }
+
+}
+
+class SupporterRatePlanItemProcessor(
+    dynamoService: SupporterDataDynamoService,
+    alarmService: AlarmService,
+    contributionProcessor: ContributionProcessor,
+) {
+
+  def processItem(supporterRatePlanItem: SupporterRatePlanItem) =
+    contributionProcessor
+      .addAmountIfContribution(supporterRatePlanItem)
+      .flatMap(writeItemToDynamo)
+
+  def writeItemToDynamo(supporterRatePlanItem: SupporterRatePlanItem) = {
+    SafeLogger.info(s"Attempting to write item to Dynamo - ${supporterRatePlanItem.asJson.noSpaces}")
+    try {
+      dynamoService.writeItem(supporterRatePlanItem).map { _ =>
+        SafeLogger.info(s"Successfully wrote item to Dynamo - ${supporterRatePlanItem.asJson.noSpaces}")
+        ()
+      }
+    } catch {
+      case t: Throwable =>
+        SafeLogger.error(scrub"Error writing item to Dynamo - ${supporterRatePlanItem.asJson.noSpaces}", t)
+        alarmService.triggerDynamoWriteAlarm match {
+          case Success(_) => Future.successful(())
+          case Failure(exception) =>
+            SafeLogger.error(scrub"Failed to trigger dynamo alarm", exception)
+            Future.successful(())
+        }
+    }
+  }
+}
+
+object SupporterRatePlanItemCodec {
+  implicit val codec: Codec[SupporterRatePlanItem] = deriveCodec
+  implicit val contributionCodec: Codec[ContributionAmount] = deriveCodec
+}
+
+object ContributionIds {
+  def forStage(stage: Stage) = stage match {
+    case PROD => List("2c92a0fc5aacfadd015ad24db4ff5e97", "2c92a0fc5e1dc084015e37f58c200eea")
+    case UAT => List("2c92c0f85ab269be015acd9d014549b7", "2c92c0f95e1d5c9c015e38f8c87d19a1")
+    case DEV => List("2c92c0f85a6b134e015a7fcd9f0c7855", "2c92c0f85e2d19af015e3896e824092c")
```

Comment:
the ids proliferate! these are rate plan ids specifically.

I've recently done some work to extract the ones we have in support-service-lambdas, but that doesn't necessarily help support-frontend or elsewhere https://github.com/guardian/support-service-lambdas/pull/1698